ni/measurementlink-python
# MeasurementService: Use a context manager to simplify ctrl-C handling

### What does this Pull Request accomplish?

Make `MeasurementService` implement the context manager protocol so that clients can use `with measurement_service.host_service():` to clean up automatically.

Remove Win32-specific ctrl-C handling. A with-statement does the right thing when Python raises `KeyboardInterrupt`.

### Why should this Pull Request be merged?

This partially fixes https://github.com/ni/measurement-services-python/issues/38 by removing Win32-specific cleanup code.

### What testing has been done?

Manually tested `sample_measurement`:
- Successfully register service and press enter
- Successfully register service and press ctrl-C
- Fail to register service -> no unexpected errors
- Unmodified examples still work when you press enter, but they no longer clean up when you press ctrl-C.
